### ISSUE TITLE
Transformer V2 @auth connection fields broken with groupClaims

### ISSUE THREAD

**Issue by alexboulay** [aws-amplify/amplify-cli#10017]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

15.14

### Amplify CLI Version

7.6.25

### What operating system are you using?

MacOS

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

api

### Amplify Commands

Not applicable

### Describe the bug

First refer to the schema to understand the queries that will be made.

Querying our models with IAM, as expected:
<img width="1799" alt="image" src="https://user-images.githubusercontent.com/70296188/159027651-5e11bd6d-4e9b-496b-ae65-aa90f56d1588.png">

Querying our models with OIDC, with the following claims in our token:
`"<domain>/owner": ["test-1"]`
Along with my sub: `oauth2|github-3|70296188`

<img width="1812" alt="image" src="https://user-images.githubusercontent.com/70296188/159028057-6e2f26e1-a107-4e37-aef9-d70e38fc0a80.png">

As you can see, the ContactInformation that are returned are the ones where ownerID is equal to my users sub.

I rather expect, that the ContactInformation would be the ones where ownerID equals `test-1`.

### Expected behavior

ContactInformation that are owned by `test-1`

### Reproduction steps

1. Follow the given schema
2. Create the models ( an org, a user, 1-2 contact information for both the user and the org )
3. Query `getOrganization` with OIDC by requesting ContactInformation and observe that the instances that are returned are not the ones owned by the queried organization

### GraphQL schema(s)

Context:

A ContactInformation can be owned either by an Organization or a User. 

```graphql
type ContactInformation @model(subscriptions: null)
  @auth(rules: [
    { allow: public, provider: iam }, 
    {
      allow: owner,
      provider: oidc,
      ownerField: "ownerID",
      identityClaim: "sub",
      operations: [ create, read, delete ]
    },
    { 
      allow: groups,
      provider: oidc,
      groupsField: "ownerID",
      groupClaim: "<domain>/owner",
      operations: [ create, read, delete ]
    },
    { 
      allow: groups,
      provider: oidc,
      groupsField: "ownerID",
      groupClaim: "<domain>/admin",
      operations: [ create, read, delete ]
    }
  ]) {
  id: ID!
  value: String!
  ownerID: ID! @index(name: "byOwner", queryField: "listContactInformationByOwner")
}
```

```gql
type Organization @model(subscriptions: null)
  @auth(rules: [
  { allow: public, provider: iam },
  { allow: private, provider: oidc } ## TEMPORARY(waiting on #9852)
  ]) {
  id: ID!
  name: String
  memberships: [Membership] @hasMany(indexName: "byGroup", fields: ["id"])
  contactInformation: [ContactInformation] @hasMany(indexName: "byOwner", fields: ["id"])
}
```
```gql
type User @model(subscriptions: null)
  @auth(rules: [
    { allow: public, provider: iam }, 
    {
      allow: owner,
      provider: oidc,
      ownerField: "id",
      identityClaim: "sub",
      operations: [ read, update ]
    }
  ]) {
  id: ID!
  contactInformation: [ContactInformation] @hasMany(indexName: "byOwner", fields: ["id"])
}
```

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

Looking into `OrganizationcontactInformationauth0Function` VTL Resolver, I have noticed two things that might be wrong:

<img width="1466" alt="amplify" src="https://user-images.githubusercontent.com/70296188/159029811-17cc4b72-aa34-4832-a88b-e4c3c4f35bc6.png">

- At line 28, the source ID value is changed to my sub which might be causing the problem where the ContactInformation returned are the ones I own (where ownerID=my sub)
- My group claims are list of strings but it looks like they are not treated as such. For exemple, at line 33, we compare a string with a list of strings.

**Comment by josefaidt** [comment]
Hey @alexboulay :wave: thanks for raising this! This appears to be a duplicate of https://github.com/aws-amplify/amplify-cli/issues/9852, marking as a duplicate bug and closing in favor of tracking the existing item 🙂 please subscribe to the linked issue for further updates

**Comment by sebastien-blain** [comment]
@josefaidt 
I do not think that #9852 refers to the same problem described here. 

Both issues share the bug of comparing the groupclaim that is a list to a string. But the new element brought here is that context.source.id is replaced by the sub in the case of a Owner @auth which then does not resolve the type correctly.
![image](https://user-images.githubusercontent.com/43256461/159039886-b25d5429-f74b-4717-9856-c32677f15951.png)

**Comment by josefaidt** [comment]
Hey @sebastien-blain great callout! With that note I'll re-open this issue and remove the `duplicate` label

**Comment by lseemann** [comment]
We are experiencing this issue, too. It’s having some very surprising and unwanted effects on the front-end. Our remedy has been to edit the put assignment, from

`$util.qr($ctx.source.put("assignedToEmployerID", $primaryRole0))`

to 

`$util.qr($ctx.source.put("assignedToEmployerID", "___xamznone___"))`

But in our situation there's only one of these puts. Looks like the OP’s resolver has a series of three, so this would probably lead to false nulls for them.
